From 4515536778a831cf5d6be86358a98f4daa6fc2a4 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Thu, 4 Apr 2019 13:45:55 +0800
Subject: [PATCH 02/30] cpufreq: imx8: delete validation of the frequency table
 from ->init()

In upstream commit d417e0691ac0 ("cpufreq: Validate frequency table in the core"),
validation of the frequency table has been moved in the cpufreq core, then in
upstream commit 2dd0df8472ff ("cpufreq: Drop cpufreq_table_validate_and_show()"),
the function is dropped. So just set policy->freq_table field directly.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/cpufreq/imx8-cpufreq.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/drivers/cpufreq/imx8-cpufreq.c b/drivers/cpufreq/imx8-cpufreq.c
index 5cefb4500303..5b998b5f306f 100644
--- a/drivers/cpufreq/imx8-cpufreq.c
+++ b/drivers/cpufreq/imx8-cpufreq.c
@@ -109,12 +109,7 @@ static int imx8_cpufreq_init(struct cpufreq_policy *policy)
 	 */
 	cpumask_copy(policy->cpus, topology_core_cpumask(policy->cpu));
 
-	ret = cpufreq_table_validate_and_show(policy, freq_table[cluster_id]);
-	if (ret) {
-		pr_err("%s: invalid frequency table: %d\n", __func__, ret);
-		return ret;
-	}
-
+	policy->freq_table = freq_table[cluster_id];
 	policy->cpuinfo.transition_latency = transition_latency[cluster_id];
 	policy->suspend_freq = policy->max;
 
-- 
2.17.1

